rs/nns/governance/src/proposals/update_canister_settings.rs

```diff
@@ -154,16 +204,55 @@ mod tests {
             },
             vec!["at least one setting", "provided"],
         );
+
+        is_invalid_proposal_with_keywords(
+            UpdateCanisterSettings {
+                settings: Some(CanisterSettings {
+                    log_visibility: Some(4),
+                    ..Default::default()
+                }),
+                ..valid_update_canister_settings.clone()
+            },
+            vec!["invalid log visibility", "4"],
+        );
+
+        is_invalid_proposal_with_keywords(
+            UpdateCanisterSettings {
+                settings: Some(CanisterSettings {
+                    log_visibility: Some(LogVisibility::Unspecified as i32),
+                    ..Default::default()
+                }),
+                ..valid_update_canister_settings.clone()
+            },
+            vec!["invalid log visibility", "0"],
+        );
+
+        is_invalid_proposal_with_keywords(
+            UpdateCanisterSettings {
+                canister_id: Some(ROOT_CANISTER_ID.get()),
+                ..valid_update_canister_settings.clone()
+            },
+            vec!["root canister", "not supported"],
+        );
     }
 
     #[cfg(feature = "test")]
     #[test]
     fn test_update_ledger_canister_settings() {
+        use candid::Decode;
+        use ic_nns_constants::GOVERNANCE_CANISTER_ID;
+
         let update_ledger_canister_settings = UpdateCanisterSettings {
             canister_id: Some(LEDGER_CANISTER_ID.get()),
             settings: Some(CanisterSettings {
-                memory_allocation: Some(1 << 30),
-                ..Default::default()
+                controllers: Some(Controllers {
+                    controllers: vec![GOVERNANCE_CANISTER_ID.get()],
+                }),
+                memory_allocation: Some(1 << 32),
+                wasm_memory_limit: Some(1 << 31),
+                compute_allocation: Some(10),
+                freezing_threshold: Some(100),
```

Comment:
NCR

I guess these numbers are arbitrary? They look like they might be special. Perhaps, add a comment that makes this fact explicit. Personally, I have a little script that I run whenever I need a not special integer test value. Or, I use 42, 43, 44, etc. It might also make sense to use 0xDEAD_BEEF when you do do not want DEADBEEF to appear in the result.

Reply:
Added a comment saying they are arbitrary